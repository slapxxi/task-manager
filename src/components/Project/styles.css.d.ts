export const name: string;
export const title: string;
