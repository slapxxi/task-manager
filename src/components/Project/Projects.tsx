import { Project as IProject } from '@local/types';
import React from 'react';
import Project from './Project';

interface Props {
  projects: IProject[];
}

function Projects({ projects }: Props) {
  return (
    <div>
      {projects.map((project) => (
        <Project project={project} key={project.id} />
      ))}
    </div>
  );
}

export default Projects;
