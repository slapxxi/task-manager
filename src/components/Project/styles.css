.name {
  font-size: 32px;
  font-weight: bold;
}
