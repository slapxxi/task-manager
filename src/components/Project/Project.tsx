import { getProjectProgress } from '@lib';
import { ProgressCheckbox, TextArea } from '@local/components';
import Button from '@local/components/Button/Button';
import IconSystem from '@local/components/IconSystem/IconSystem';
import { Project as IProject } from '@local/types';
import React from 'react';
import styles from './styles.css';

interface Props {
  project: IProject;
  children?: React.ReactNode;
  onEdit?: (project: IProject) => void;
}

function Project({ project, onEdit, children }: Props) {
  if (project === undefined) {
    return null;
  }

  function handleChange(value: string) {
    if (onEdit) {
      onEdit({ ...project, name: value });
    }
  }

  return (
    <div>
      <div className={styles.title}>
        <ProgressCheckbox size={22} progress={getProjectProgress(project)} />
        <TextArea
          className={styles.name}
          placeholder="Project Name..."
          value={project.name}
          onChange={handleChange}
          data-testid="project-name"
        />
        <Button>
          <IconSystem name="dots" size={22} />
        </Button>
      </div>
      <div className={styles.subtitle}>{project.tasks.length} tasks</div>
      {children}
    </div>
  );
}

export default Project;
