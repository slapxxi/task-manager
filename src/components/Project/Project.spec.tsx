// @ts-ignore
import React from 'react';
import { fireEvent, render } from 'react-testing-library';
import Project from './Project';

jest.mock('@local/components/IconSystem/IconSystem', () => () => 'icon');
jest.mock('@local/hooks/useWorker');

const project = {
  id: 'test',
  name: 'Test Project',
  tasks: [],
};

it('renders project', () => {
  const { container } = render(<Project project={project} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('calls `onEdit` when project is modified', () => {
  const spy = jest.fn();
  const { getByTestId } = render(<Project project={project} onEdit={spy} />);
  fireEvent.change(getByTestId('project-name'), {
    target: { value: 'New Project' },
  });
  expect(spy).toHaveBeenCalledTimes(1);
  expect(spy).toHaveBeenCalledWith({
    id: 'test',
    name: 'New Project',
    tasks: [],
  });
});
