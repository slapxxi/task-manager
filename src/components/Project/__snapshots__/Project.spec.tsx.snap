// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders project 1`] = `
<div>
  <div
    class="title"
  >
    <div>
      <div
        class="container"
        style="height: 22px;"
      >
        <input
          class="input"
          type="checkbox"
        />
        <svg
          height="22"
          viewBox="0 0 100 100"
          width="22"
        >
          <circle
            cx="50"
            cy="50"
            fill="none"
            r="44"
            stroke="#99a"
            stroke-width="8"
          />
          <path
            d="M 50 17 A 33 33 0 0 0 50 17 L 50 50 L 50 17 Z"
            fill="#aab"
            stroke="none"
            stroke-width="6"
          />
          /&gt;
        </svg>
      </div>
    </div>
    <textarea
      class="name sc-bwzfXH evfpOC"
      data-testid="project-name"
      placeholder="Project Name..."
      rows="1"
      style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
    >
      Test Project
    </textarea>
    <button
      class="sc-bdVaJa fFWIpk"
    >
      icon
    </button>
  </div>
  <div
    class="subtitle"
  >
    0
     tasks
  </div>
</div>
`;
